Vault/Library/CachePersistence.cs

```diff
@@ -89,17 +91,15 @@ public void Rename(string newName, AuthenticationRecord authRecord)
             BeforeAccessNotification(null);
         }
 
-        /// <summary>
-        /// Triggered right before MSAL needs to access the cache
-        /// Reload the cache from the persistent store in case it changed since the last access
-        /// </summary>
         void BeforeAccessNotification(AuthenticationRecord authRecord)
         {
             lock (FileLock)
             {
                 // Load the previously serialized AuthenticationRecord from disk and deserialize it.
                 var authRecordStream = new FileStream(FileName, FileMode.Open, FileAccess.Read);
-                authRecord = AuthenticationRecord.Deserialize(authRecordStream);
+                byte[] decryptData = DecryptDataFromStream(entropy, DataProtectionScope.CurrentUser, authRecordStream, bytesWritten);
+                MemoryStream memoryStream = new MemoryStream(decryptData);

```

Comment:
```suggestion
                using MemoryStream memoryStream = new MemoryStream(decryptData);

```

You should always dispose disposables when finished.

Reply:
Updated according to your comments.
Thanks for your guidance :)